docs/pages/database-access/guides/clickhouse-self-hosted.mdx

```diff
@@ -0,0 +1,305 @@
+---
+title: Database Access with ClickHouse
+description: How to configure Teleport database access with ClickHouse.
+---
+
+The Teleport Clickhouse integration allows you to enroll ClickHouse databases
+with Teleport. 
+
+The Teleport Database Service authenticates to ClickHouse using x509
+certificates, which are available for the ClickHouse HTTP and native TCP
+[interfaces](https://clickhouse.com/docs/en/interfaces/overview). The Teleport
+Database Service can communicate in both the ClickHouse TCP and HTTP protocols,
+and you can select which protocol to use when configuring the Teleport Database
+Service.
+
+Teleport audit logs for query activity are only supported for the ClickHouse
+HTTP interface. Teleport support for ClickHouse's native interfaces does not
+include audit logs for database query activity.
+
+<Admonition type="warning" title="Preview">
+
+ Teleport support for ClickHouse is currently in Preview.
+
+</Admonition>
+
+This guide will help you to:
+
+- Install and configure Teleport.
+- Set up Teleport to access your self-hosted ClickHouse database.
+- Connect to your database through Teleport.
+
+<ScopedBlock scope={["oss", "enterprise"]}>
+![Teleport Database Access Self-hosted ClickHouse](../../../img/database-access/guides/clickhouse_selfhosted_selfhosted.png)
+</ScopedBlock>
+<ScopedBlock scope={["cloud"]}>
+![Teleport Database Access ClickHouse Cloud](../../../img/database-access/guides/clickhouse_selfhosted_cloud.png)
+</ScopedBlock>
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- Either a Linux host or Kubernetes cluster where you will run the Teleport Database Service.
+
+You will also need the following, depending on the ClickHouse protocol you
+choose:
+
+<Tabs>
+<TabItem label="HTTP">
+- A self-hosted deployment of ClickHouse Server v22.3 or later.
+</TabItem>
+<TabItem label="Native TCP">
+- A self-hosted deployment of ClickHouse Server v23.3 or later.
+- The [clickhouse-client](https://clickhouse.com/docs/en/interfaces/cli) installed and added to your system's `PATH` environment variable.
+</TabItem>
+</Tabs>
+
+## Step 1/5. Create a Teleport token and user
+
+(!docs/pages/includes/database-access/token.mdx!)
+
+(!docs/pages/includes/database-access/create-user.mdx!)
```

Comment:
Similar to L28, to keep this page specific to the topic at hand, would it make sense to remove user creation and just include "a user with the access role" in the prerequisites?

Reply:
This is common steps for all db guides so I think that it makes sense but requires a generic alignment in all db guides. I will create a follow up issue.